// @pothos/core/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
var _exportNames = {
    FieldBuilder: true,
    RootFieldBuilder: true,
    QueryFieldBuilder: true,
    MutationFieldBuilder: true,
    SubscriptionFieldBuilder: true,
    ObjectFieldBuilder: true,
    InterfaceFieldBuilder: true,
    InputFieldBuilder: true,
    BaseTypeRef: true,
    EnumRef: true,
    InputObjectRef: true,
    InterfaceRef: true,
    ObjectRef: true,
    ScalarRef: true,
    UnionRef: true,
    BuildCache: true,
    BuiltinScalarRef: true,
    FieldRef: true,
    InputTypeRef: true,
    InputFieldRef: true,
    ImplementableInputObjectRef: true,
    ImplementableInterfaceRef: true,
    ImplementableObjectRef: true,
    OutputTypeRef: true
};
Object.defineProperty(exports, "BuildCache", {
    enumerable: true,
    get: function() {
        return _buildCache.default;
    }
});
Object.defineProperty(exports, "BuiltinScalarRef", {
    enumerable: true,
    get: function() {
        return _builtinScalar.default;
    }
});
Object.defineProperty(exports, "FieldRef", {
    enumerable: true,
    get: function() {
        return _field.default;
    }
});
Object.defineProperty(exports, "InputTypeRef", {
    enumerable: true,
    get: function() {
        return _input1.default;
    }
});
Object.defineProperty(exports, "InputFieldRef", {
    enumerable: true,
    get: function() {
        return _inputField.default;
    }
});
Object.defineProperty(exports, "ImplementableInputObjectRef", {
    enumerable: true,
    get: function() {
        return _inputObject.ImplementableInputObjectRef;
    }
});
Object.defineProperty(exports, "ImplementableInterfaceRef", {
    enumerable: true,
    get: function() {
        return _interface1.ImplementableInterfaceRef;
    }
});
Object.defineProperty(exports, "ImplementableObjectRef", {
    enumerable: true,
    get: function() {
        return _object1.ImplementableObjectRef;
    }
});
Object.defineProperty(exports, "OutputTypeRef", {
    enumerable: true,
    get: function() {
        return _output.default;
    }
});
exports.UnionRef = exports.ScalarRef = exports.ObjectRef = exports.InterfaceRef = exports.InputObjectRef = exports.EnumRef = exports.BaseTypeRef = exports.InputFieldBuilder = exports.InterfaceFieldBuilder = exports.ObjectFieldBuilder = exports.SubscriptionFieldBuilder = exports.MutationFieldBuilder = exports.QueryFieldBuilder = exports.RootFieldBuilder = exports.FieldBuilder = exports.default = void 0;
require("./types/global");
var _builder = _interopRequireDefault(require("./builder"));
var _builder1 = _interopRequireDefault(require("./fieldUtils/builder"));
var _input = _interopRequireDefault(require("./fieldUtils/input"));
var _interface = _interopRequireDefault(require("./fieldUtils/interface"));
var _mutation = _interopRequireDefault(require("./fieldUtils/mutation"));
var _object = _interopRequireDefault(require("./fieldUtils/object"));
var _query = _interopRequireDefault(require("./fieldUtils/query"));
var _root = _interopRequireDefault(require("./fieldUtils/root"));
var _subscription = _interopRequireDefault(require("./fieldUtils/subscription"));
var _base = _interopRequireDefault(require("./refs/base"));
var _enum = _interopRequireDefault(require("./refs/enum"));
var _inputObject = _interopRequireWildcard(require("./refs/input-object"));
var _interface1 = _interopRequireWildcard(require("./refs/interface"));
var _object1 = _interopRequireWildcard(require("./refs/object"));
var _scalar = _interopRequireDefault(require("./refs/scalar"));
var _union = _interopRequireDefault(require("./refs/union"));
var _plugins = _interopRequireWildcard(require("./plugins"));
Object.keys(_plugins).forEach(function(key) {
    if (key === "default" || key === "__esModule") return;
    if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
    if (key in exports && exports[key] === _plugins[key]) return;
    Object.defineProperty(exports, key, {
        enumerable: true,
        get: function() {
            return _plugins[key];
        }
    });
});
var _types = _interopRequireWildcard(require("./types"));
Object.keys(_types).forEach(function(key) {
    if (key === "default" || key === "__esModule") return;
    if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
    if (key in exports && exports[key] === _types[key]) return;
    Object.defineProperty(exports, key, {
        enumerable: true,
        get: function() {
            return _types[key];
        }
    });
});
var _utils = _interopRequireWildcard(require("./utils"));
Object.keys(_utils).forEach(function(key) {
    if (key === "default" || key === "__esModule") return;
    if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
    if (key in exports && exports[key] === _utils[key]) return;
    Object.defineProperty(exports, key, {
        enumerable: true,
        get: function() {
            return _utils[key];
        }
    });
});
var _buildCache = _interopRequireDefault(require("./build-cache"));
var _builtinScalar = _interopRequireDefault(require("./refs/builtin-scalar"));
var _field = _interopRequireDefault(require("./refs/field"));
var _input1 = _interopRequireDefault(require("./refs/input"));
var _inputField = _interopRequireDefault(require("./refs/input-field"));
var _output = _interopRequireDefault(require("./refs/output"));
function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache() {
    if (typeof WeakMap !== "function") return null;
    var cache = new WeakMap();
    _getRequireWildcardCache = function() {
        return cache;
    };
    return cache;
}
function _interopRequireWildcard(obj) {
    if (obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache();
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {};
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
const SchemaBuilder = _builder.default;
var _default = SchemaBuilder;
exports.default = _default;
const FieldBuilder = _builder1.default;
exports.FieldBuilder = FieldBuilder;
const RootFieldBuilder = _root.default;
exports.RootFieldBuilder = RootFieldBuilder;
const QueryFieldBuilder = _query.default;
exports.QueryFieldBuilder = QueryFieldBuilder;
const MutationFieldBuilder = _mutation.default;
exports.MutationFieldBuilder = MutationFieldBuilder;
const SubscriptionFieldBuilder = _subscription.default;
exports.SubscriptionFieldBuilder = SubscriptionFieldBuilder;
const ObjectFieldBuilder = _object.default;
exports.ObjectFieldBuilder = ObjectFieldBuilder;
const InterfaceFieldBuilder = _interface.default;
exports.InterfaceFieldBuilder = InterfaceFieldBuilder;
const InputFieldBuilder = _input.default;
exports.InputFieldBuilder = InputFieldBuilder;
const BaseTypeRef = _base.default;
exports.BaseTypeRef = BaseTypeRef;
const EnumRef = _enum.default;
exports.EnumRef = EnumRef;
const InputObjectRef = _inputObject.default;
exports.InputObjectRef = InputObjectRef;
const InterfaceRef = _interface1.default;
exports.InterfaceRef = InterfaceRef;
const ObjectRef = _object1.default;
exports.ObjectRef = ObjectRef;
const ScalarRef = _scalar.default;
exports.ScalarRef = ScalarRef;
const UnionRef = _union.default;
exports.UnionRef = UnionRef;
