// @pothos/core/lib/types/type-params.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.typeBrandKey = exports.outputFieldShapeKey = exports.inputFieldShapeKey = exports.inputShapeKey = exports.abstractReturnShapeKey = exports.parentShapeKey = exports.outputShapeKey = void 0;
const outputShapeKey = Symbol.for('Pothos.outputShapeKey');
exports.outputShapeKey = outputShapeKey;
const parentShapeKey = Symbol.for('Pothos.parentShapeKey');
exports.parentShapeKey = parentShapeKey;
const abstractReturnShapeKey = Symbol.for('Pothos.abstractReturnShapeKey');
exports.abstractReturnShapeKey = abstractReturnShapeKey;
const inputShapeKey = Symbol.for('Pothos.inputShapeKey');
exports.inputShapeKey = inputShapeKey;
const inputFieldShapeKey = Symbol.for('Pothos.inputFieldShapeKey');
exports.inputFieldShapeKey = inputFieldShapeKey;
const outputFieldShapeKey = Symbol.for('Pothos.outputFieldShapeKey');
exports.outputFieldShapeKey = outputFieldShapeKey;
const typeBrandKey = Symbol.for('Pothos.typeBrandKey');
exports.typeBrandKey = typeBrandKey;
