// @trpc/server/adapters/aws-lambda/dist/trpc-server-adapters-aws-lambda.esm.js
import { URLSearchParams } from 'url';
import '../../../dist/router-93491dae.esm.js';
import 'events';
import 'http';
import '../../../dist/nodeHTTPRequestHandler-372a4a4f.esm.js';
import { r as resolveHTTPResponse } from '../../../dist/resolveHTTPResponse-ddb56677.esm.js';
import '../../../dist/transformTRPCResponse-45c225c3.esm.js';
import '../../../dist/codes-5678cc97.esm.js';

function lambdaEventToHTTPRequest(event) {
  const query = new URLSearchParams();

  for (const [key, value] of Object.entries((_event$multiValueQuer = event.multiValueQueryStringParameters) !== null && _event$multiValueQuer !== void 0 ? _event$multiValueQuer : {})) {
    var _event$multiValueQuer;

    for (const v of value !== null && value !== void 0 ? value : []) {
      query.append(key, v);
    }
  }

  return {
    method: event.httpMethod,
    query,
    headers: event.headers,
    body: event.body
  };
}

function transformHeaders(headers) {
  const obj = {};

  for (const [key, value] of Object.entries(headers)) {
    if (typeof value === 'undefined') {
      continue;
    }

    obj[key] = Array.isArray(value) ? value.join(', ') : value;
  }

  return obj;
}

function createApiGatewayHandler(opts) {
  return async (event, context) => {
    var _response$body, _response$headers;

    const req = lambdaEventToHTTPRequest(event);
    const path = event.path;

    const createContext = async function _createContext() {
      var _opts$createContext;

      return await ((_opts$createContext = opts.createContext) === null || _opts$createContext === void 0 ? void 0 : _opts$createContext.call(opts, {
        event,
        context
      }));
    };

    const response = await resolveHTTPResponse({
      router: opts.router,
      batching: opts.batching,
      responseMeta: opts === null || opts === void 0 ? void 0 : opts.responseMeta,
      createContext,
      req,
      path,
      error: null,

      onError(o) {
        var _opts$onError;

        opts === null || opts === void 0 ? void 0 : (_opts$onError = opts.onError) === null || _opts$onError === void 0 ? void 0 : _opts$onError.call(opts, { ...o,
          req: event
        });
      }

    });
    const resp = {
      statusCode: response.status,
      body: (_response$body = response.body) !== null && _response$body !== void 0 ? _response$body : '',
      headers: transformHeaders((_response$headers = response.headers) !== null && _response$headers !== void 0 ? _response$headers : {})
    };
    return resp;
  };
}

export { createApiGatewayHandler };
